### DISCUSSION TITLE
Reading additional data from Camtasia .trec files

### DISCUSSION THREAD

**Discussion by NiallTracey** [mifi/lossless-cut#1035] [Q&A]
Hi,
Finding the software very handy -- I use it to export Camtasia screen recording+webcam to let me edit in Vegas Pro instead of Camtasia itself. (Better greenscreen, nicer annotations etc.) I use this for blended learning classes in Computer Science.

This app gives me the video and audio, but it doesn't give me a way to extract the additional data tracks: mouse cursor, key presses and markers (I like to lay down markers during recording to identify where I've fluffed a line, or where I need to insert an annotation).

I realise that's probably beyond the scope and purpose of this app, but I haven't been able to find any specs for the .trec container, and I'm not currently aware of anyone other app that imports it.

I'm guessing it's just some standard container format, simply renamed to hide it.

If you could tell me what that format is and point me to a spec, I'm sure I'd be able to extract the data tracks using Python.

Hope you can help.

Thanks either way -- this app saves me loads of time.

[Also, just to clarify -- I've tried the "Extract unprocessable tracks" in the settings menu, and this data isn't being stored in the tracks, so presumably is in the container metadata somewhere, hence me wanting to know about the container format.]

**Comment by mifi** [comment]
Hi! I don't know anything about this format unfortunately, other than the fact that ffmpeg is able to read it (and therefore also losslesscut). If ffmpeg or ffprobe cannot read the meta, maybe exiftool can?

Related: https://github.com/mifi/lossless-cut/issues/975

**Reply by NiallTracey** [reply]
Thanks -- that's useful information in and of itself, as it confirms it's just a standard container format that's been renamed with a different extension.

I'll try again with FFMPEG -- presumably last time I checked, something was configured to go by file extensions rather than reading the file to find the format.
